src/v/cloud_storage/cache_service.cc

```diff
@@ -230,8 +234,13 @@ ss::future<> cache::trim_throttled() {
 ss::future<> cache::trim() {
     vassert(ss::this_shard_id() == 0, "Method can only be invoked on shard 0");
     gate_guard guard{_gate};
-    auto [walked_cache_size, candidates_for_deletion, _]
-      = co_await _walker.walk(_cache_dir.native(), _access_time_tracker);
+    auto [walked_cache_size, filtered_out_files, candidates_for_deletion, _]
+      = co_await _walker.walk(
+        _cache_dir.native(), _access_time_tracker, [](std::string_view path) {
+            return !(
+              std::string_view(path).ends_with(".tx")
+              || std::string_view(path).ends_with(".index"));
```

Comment:
Now that we're not explicitly trimming these, we should take care to make our behavior across a crash copes well. I think right now if we delete the log file and crash before we get to the ancillary files, they'll be stuck around forever. We should probably reverse the order to delete tx and index first.

Reply:
The start-up trim does not apply the filtering precisely for this reason. If we crash in between, we'll still attempt to clean up `index` and `tx` files on start-up.